### PR TITLE
Removes `add_eval_metrics` and just puts them in the relevant constructors

### PR DESCRIPTION
**Pull request by dakinggg**
Manually tested by running mosaic gpt and gpt neo and confirming that it made it through the first eval with icl metrics.

### CODE REVIEW

File: examples/llm/src/models/hf/hf_t5.py
Comment (alextrott16) [inline examples/llm/src/models/hf/hf_t5.py:92]: These ICL eval tools don't support encoder-decoder yet. Maybe we should hold off on adding these until they are supported?

**Review by alextrott16 — APPROVED** [review]
LGTM and if it works then that seems like a reasonable test.

Out of curiosity, how does this work with eval dataloaders that don't specify the metrics that apply to them? I know that _evaluators_ let you name the relevant metrics, but that's not required, right? I thought that if you don't use an evaluator object for the eval dataloader, it will try to compute all the eval metrics. And I'd expect that to be an issue when, say, your standard LM eval dataloader doesn't produce batches that work with one of the ICL metrics.

Can you help correct my misunderstanding?

**Comment by dakinggg** [comment]
Your understanding is correct. In examples repo we make sure all eval things are `Evaluator`, so just hardcoding the metrics seemed like the cleanest solution here. This is actually just necessary given the current composer design for eval metrics, which is a bit wonky. We do have some code in `update_metric` that skips updating metrics when the batch is the wrong thing, so it still shouldn't crash in that case, but you do get some bad logging I believe, don't quite remember.

File: examples/llm/src/models/hf/hf_t5.py
Comment (dakinggg) [reply examples/llm/src/models/hf/hf_t5.py:92]: ah, good call. will remove.

**Pull request merged**
